src/main/scala/example/Task2RepartitioningApproach.scala

```diff
@@ -0,0 +1,119 @@
+    //              :           :  +- Exchange hashpartitioning(UserId#14, 200), REPARTITION_BY_COL, [plan_id=59] // REPARTITION USERS BY USER ID
+    //              :           :     +- LocalTableScan [UserId#14] // LOAD USERS
+    //              :           +- Sort [UserId#476 ASC NULLS FIRST], false, 0 // PREPARATION FOR JOIN OF ORDERS WITH USERS
+    //              :              +- Exchange hashpartitioning(UserId#476, 200), ENSURE_REQUIREMENTS, [plan_id=80]
+    //              :                 +- Union // DO UNION OF ORDERS
+    //              :                    :- Exchange hashpartitioning(UserId#476, 200), REPARTITION_BY_COL, [plan_id=61] // REPARTITION ORDERS BY USER ID
+    //              :                    :  +- LocalTableScan [UserId#476, ProductId#488, TotalValue#524] // LOAD ORDERS
+    //              :                    +- Exchange hashpartitioning(UserId#322, 200), REPARTITION_BY_COL, [plan_id=63] // REPARTITION ORDERS BY USER ID
+    //              :                       +- LocalTableScan [UserId#322, ProductId#334, TotalValue#370] // LOAD ORDERS
+    //              +- Sort [ProductId#84 ASC NULLS FIRST], false, 0 // PREPARATION FOR MAIN JOIN
+    //                +- Exchange hashpartitioning(ProductId#84, 200), ENSURE_REQUIREMENTS, [plan_id=93]
+    //                  +- Project [ProductId#84, CategoryId#92]
+    //                    +- SortMergeJoin [CategoryId#92], [CategoryId#148], LeftOuter // JOIN OF PRODUCTS WITH CATEGORIES
+    //                      :- Sort [CategoryId#92 ASC NULLS FIRST], false, 0 // PREPARATION FOR JOIN OF PRODUCTS WITH CATEGORIES
+    //                      :  +- Exchange hashpartitioning(CategoryId#92, 200), REPARTITION_BY_COL, [plan_id=68] // REPARTITION PRODUCTS BY CATEGORY ID
+    //                      :     +- LocalTableScan [ProductId#84, CategoryId#92] // LOAD PRODUCTS
+    //                      +- Sort [CategoryId#148 ASC NULLS FIRST], false, 0
+    //                         +- Exchange hashpartitioning(CategoryId#148, 200), REPARTITION_BY_COL, [plan_id=70] // REPARTITION CATEGORIES BY CATEGORY ID
+    //                            +- LocalTableScan [CategoryId#148] // LOAD CATEGORIES
+
+    // WITHOUT "COMPLICATIONS"
+    // PERFORMANCE TEST
+
+    //  val COMPLETED_ORDERS_COUNT = 50000
+    //  val NOT_COMPLETED_ORDERS_COUNT = 50000
+    //  val USERS_COUNT = 10000
+    //  val INSTANCES_COUNT = 5
+    //  val COUNTRIES_COUNT = 10
+    //  val MANUFACTURERS_COUNT = 20
+    //  val PRODUCTS_COUNT = 1000
+    //  val CATEGORIES_COUNT = 100
+    //
+    //  val PRICE_INTERVAL = 100
+    //  val MAX_PRICE = 500
+    //
+    //  val WEIGHT_INTERVAL = 20
+    //  val MAX_WEIGHT = 100
+    //
+    //  val MAX_DAYS = 28
+
+    // Around 6s, the effort is moved from the end to the beginning of computations. Repartitioning is approximately as costly as a shuffle.
```

Comment:
yeah, but what's mean?
Can you force better performance tests on the build?
How to do it?

Reply:
And what time do we spend on the start of the application?
Who to do it better?

Reply:
It was about doing some experimenting about performance, nothing related to unit tests or build.